browser/static/js/adjust.js

```diff
@@ -1,277 +1,290 @@
-}
+    postCompImg.src = this.canvas.toDataURL();
+  }
 
-// apply outlines, transparent highlighting for RGB
-function postCompAdjustRGB() {
-  let canvas = document.getElementById('hidden_seg_canvas');
-  let ctx = $('#hidden_seg_canvas').get(0).getContext('2d');
-  ctx.imageSmoothingEnabled = false;
+  // apply outlines, transparent highlighting for RGB
+  postCompAdjustRGB(contrastedRaw, postCompImg,
+      current_highlight, edit_mode, brush, mode) {
 
-  // draw seg_image so we can extract image data
-  ctx.clearRect(0, 0, rawWidth, rawHeight);
-  ctx.drawImage(contrastedRaw, 0, 0, rawWidth, rawHeight);
+    // draw contrastedRaw so we can extract image data
+    this.ctx.clearRect(0, 0, this.width, this.height);
+    this.ctx.drawImage(contrastedRaw, 0, 0, this.width, this.height);
 
-  // add outlines around conversion brush target/value
-  let imgData = ctx.getImageData(0, 0, rawWidth, rawHeight);
+    // add outlines around conversion brush target/value
+    let imgData = this.ctx.getImageData(0, 0, this.width, this.height);
 
-  let redOutline, r1, singleOutline, o1, outlineAll, translucent, t1, t2;
+    let redOutline, r1, singleOutline, o1, outlineAll, translucent, t1, t2;
 
-  // red outline for conversion brush target
-  if (edit_mode && brush.conv && brush.target !== -1) {
-    redOutline = true;
-    r1 = brush.target;
-  }
+    // red outline for conversion brush target
+    if (edit_mode && brush.conv && brush.target !== -1) {
+      redOutline = true;
+      r1 = brush.target;
+    }
 
-  // singleOutline never on for RGB
+    // singleOutline never on for RGB
 
-  outlineAll = true;
+    outlineAll = true;
 
-  // translucent highlight
-  if (current_highlight) {
-    translucent = true;
-    if (edit_mode) {
-      t1 = brush.value;
-    } else {
-      t1 = mode.highlighted_cell_one;
-      t2 = mode.highlighted_cell_two;
+    // translucent highlight
+    if (current_highlight) {
+      translucent = true;
+      if (edit_mode) {
+        t1 = brush.value;
+      } else {
+        t1 = mode.highlighted_cell_one;
+        t2 = mode.highlighted_cell_two;
+      }
     }
-  }
 
-  postCompositeLabelMod(imgData, redOutline, r1, singleOutline, o1,
-    outlineAll, translucent, t1, t2);
+    this.postCompositeLabelMod(imgData, redOutline, r1, singleOutline, o1,
+      outlineAll, translucent, t1, t2);
 
-  ctx.putImageData(imgData, 0, 0);
-
-  postCompImg.src = canvas.toDataURL();
-}
+    this.ctx.putImageData(imgData, 0, 0);
 
-function prepareRaw() {
-  contrastRaw(current_contrast, brightness);
-}
+    postCompImg.src = this.canvas.toDataURL();
+  }
 
-function segAdjust() {
-  segLoaded = true;
-  if (rawLoaded && segLoaded) {
-    if (rgb) {
-      postCompAdjustRGB();
-    } else {
-      compositeImages();
+  // TODO: not sure how to scope segLoaded and rawLoaded.
+  // they are changed by these adjust functions so the image composite
+  // will only be generated when both source images are ready to use
```

Comment:
It looks like `segLoaded` and `rawLoaded` are only used in these 2 methods. Can these functions get pulled out of the class and moved into the place where it's being called? That would remove the troublesome variables from `adjust.js` completely.

Reply:
Yes--I think it'll work to make `segLoaded` an attribute of the `ImageAdjuster` class and set it to false at the start of `preCompAdjust`. (Similarly with `rawLoaded`.) Then the only place in the main js file that it needs to be explicitly updated is when new images are loaded (changed frames, modified annotations, changed channels).